medrag_multi_modal/semantic_chunking.py

```diff
@@ -108,12 +116,23 @@ async def process_document(idx, document):
 
         async def process_all_documents():
             tasks = []
+            streamlit_progressbar = (
+                st.progress(0, text="Chunking documents")
+                if self.streamlit_mode
+                else None
+            )
             for idx, document in track(
                 enumerate(document_dataset),
                 total=len(document_dataset),
                 description="Chunking documents",
             ):
                 tasks.append(process_document(idx, document))
+                if streamlit_progressbar:
+                    progress_percentage = int((idx / (len(document_dataset) - 1)) * 100)
```

Comment:
```suggestion
                    progress_percentage = int((idx / (len(document_dataset))) * 100)
```

The progress bars in all the cases feel like they are lagging one step behind.
can be a future improvement.